core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MergeCustomHeadersStage.java

```diff
@@ -52,7 +54,15 @@ public MergeCustomHeadersStage(HttpClientDependencies dependencies) {
     private final Map<String, List<String>> mergeHeaders(Map<String, List<String>>... headers) {
         Map<String, List<String>> result = new LinkedHashMap<>();
         for (Map<String, List<String>> header : headers) {
-            header.forEach((k, v) -> result.computeIfAbsent(k, ignored -> new ArrayList<>()).addAll(v));
```

Comment:
We're now introducing an implicit 'priority' in the sense that the last header 'wins'. Can we add a comment about that?

Reply:
Sure, i'll add a note on the setters in the Client/Request override configuration.